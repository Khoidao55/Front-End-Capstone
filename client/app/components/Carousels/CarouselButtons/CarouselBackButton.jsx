import React from 'react';
import propTypes from 'proptypes';
import './CarouselButtons.css';

const CarouselBackButton = ({ atStart, handleBack }) => {
  const buttonVisibility = atStart ? 'carousel-btn-invisible' : 'carousel-btn-visible';

  return (
    <button
      aria-label="Carousel Back Button"
      type="button"
      className={`carousel-btn-back ${buttonVisibility}`}
      onClick={handleBack}
    >
      <i className="fas fa-arrow-left" />
    </button>
  );
};

CarouselBackButton.propTypes = {
  atStart: propTypes.bool.isRequired,
  handleBack: propTypes.func.isRequired,
};

export default CarouselBackButton;
