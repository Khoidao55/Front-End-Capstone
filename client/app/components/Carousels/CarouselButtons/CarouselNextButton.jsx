import React from 'react';
import propTypes from 'proptypes';
import './CarouselButtons.css';

const CarouselNextButton = ({ atEnd, handleNext }) => {
  const buttonVisibility = atEnd ? 'carousel-btn-invisible' : 'carousel-btn-visible';

  return (
    <button
      aria-label="Carousel Next Button"
      type="button"
      className={`carousel-btn-next ${buttonVisibility}`}
      onClick={handleNext}
    >
      <i className="fas fa-arrow-right" />
    </button>
  );
};

CarouselNextButton.propTypes = {
  atEnd: propTypes.bool.isRequired,
  handleNext: propTypes.func.isRequired,
};

export default CarouselNextButton;
