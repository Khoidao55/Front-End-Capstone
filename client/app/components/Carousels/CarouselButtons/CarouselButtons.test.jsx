import React from 'react';
import { shallow } from 'enzyme';
import CarouselBackButton from './CarouselBackButton.jsx';
import CarouselNextButton from './CarouselNextButton.jsx';

const dummyBack = {
  atStart: false,
  handleBack: () => { console.log('click back!'); },
};

const dummyNext = {
  atEnd: false,
  handleNext: () => { console.log('click next!'); },
};

describe('Carousel Buttons', () => {
  test('back button renders correctly', () => {
    const wrapper = shallow(<CarouselBackButton { ...dummyBack } />);

    expect(wrapper).toMatchSnapshot();
  });

  test('next button renders correctly', () => {
    const wrapper = shallow(<CarouselNextButton { ...dummyNext } />);

    expect(wrapper).toMatchSnapshot();
  });
});
