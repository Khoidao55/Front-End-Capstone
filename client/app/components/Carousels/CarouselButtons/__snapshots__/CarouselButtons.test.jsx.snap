// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Buttons back button renders correctly 1`] = `ShallowWrapper {}`;

exports[`Carousel Buttons next button renders correctly 1`] = `ShallowWrapper {}`;
